### PR TITLE
strawberry: init at 0.6.3

### PR DESCRIPTION
**Pull request by peterhoeg**
##### Motivation for this change

strawberry is a *maintained* fork of clementine. Can highly recommend it.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @ttuegel as clementine maintainer

### CODE REVIEW

File: pkgs/applications/audio/strawberry/default.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{ mkDerivation, stdenv, lib, fetchFromGitHub, cmake, pkgconfig
+, alsaLib, boost, chromaprint, fftw, gnutls, libcdio, libmtp, libpthreadstubs, libtasn1, libXdmcp, pcre, protobuf, sqlite, taglib
```
Comment (alexarice) [inline pkgs/applications/audio/strawberry/default.nix]: This line is quite long, could it be split into two?

File: pkgs/applications/audio/strawberry/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
+{ mkDerivation, stdenv, lib, fetchFromGitHub, cmake, pkgconfig
```
Comment (alexarice) [inline pkgs/applications/audio/strawberry/default.nix]: I've never seen mkDerivation as a function arg but I guess it makes sense

Comment (alexarice) [reply pkgs/applications/audio/strawberry/default.nix]: Though this does mean `stdenv` has not been used anywhere

File: pkgs/applications/audio/strawberry/default.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  src = fetchFromGitHub {
+    owner = "jonaski";
+    repo = pname;
+    rev = version;
+    sha256 = "01j5jzzicy895kg9sjy46lbcm5kvf3642d3q5wwb2fyvyq1fbcv0";
+  };
+
+  buildInputs = [
+    alsaLib boost chromaprint fftw gnutls libcdio libmtp libpthreadstubs libtasn1 libXdmcp pcre protobuf sqlite taglib
```
Comment (alexarice) [inline pkgs/applications/audio/strawberry/default.nix]: this line is also quite long

File: pkgs/applications/audio/strawberry/default.nix [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    qtbase qtx11extras
+  ] ++ lib.optionals stdenv.isLinux [
+    libpulseaudio libselinux libsepol p11_kit utillinux
+  ] ++ lib.optionals withVlc [
```
Comment (alexarice) [inline pkgs/applications/audio/strawberry/default.nix]: `lib.optional withVlc vlc` would work here

File: pkgs/applications/audio/strawberry/default.nix (hunk 1 above)
Comment (peterhoeg) [reply pkgs/applications/audio/strawberry/default.nix]: I've just run nixpkgs-fmt on it to clean things up.

File: pkgs/applications/audio/strawberry/default.nix (hunk 2 above)
Comment (peterhoeg) [reply pkgs/applications/audio/strawberry/default.nix]: See https://github.com/NixOS/nixpkgs/issues/65399.

stdenv is used in ```stdenv.isLinux```.

File: pkgs/applications/audio/strawberry/default.nix (hunk 4 above)
Comment (peterhoeg) [reply pkgs/applications/audio/strawberry/default.nix]: Done.

File: pkgs/applications/audio/strawberry/default.nix (hunk 2 above)
Comment (alexarice) [reply pkgs/applications/audio/strawberry/default.nix]: of course, my bad

File: pkgs/applications/audio/strawberry/default.nix (hunk 1 above)
Comment (alexarice) [reply pkgs/applications/audio/strawberry/default.nix]: I wasn't aware nixpkgs-fmt was released yet.

**Comment by alexarice** [comment]
I would run nix-review but it's trying to run a mass rebuild so I might try again when nixpkgs moves forwards

File: pkgs/applications/audio/strawberry/default.nix (hunk 2 above)
Comment (alexarice) [reply pkgs/applications/audio/strawberry/default.nix]: Thanks for the link as well

File: pkgs/applications/audio/strawberry/default.nix (hunk 1 above)
Comment (peterhoeg) [reply pkgs/applications/audio/strawberry/default.nix]: It's in unstable.

**Comment by alexarice** [comment]
I gave nix-review another try but I think it'd take all day on my laptop.

While running it I did get the following errors though they don't seem related to this package so I'm a bit confused.

```
builder for '/nix/store/7sfljq22sax1spzlh1jk2kpkzy5s8ncf-breeze-qt5-5.15.5.drv' failed with exit code 2; last 10 log lines:
  Scanning dependencies of target breeze-settings5
  [ 10%] Building CXX object kstyle/config/CMakeFiles/breeze-settings5.dir/breeze-settings5_autogen/mocs_compilation.cpp.o
  [ 11%] Building CXX object kstyle/config/CMakeFiles/breeze-settings5.dir/main.cpp.o
  [ 12%] Linking CXX executable breeze-settings5
  /nix/store/mgdjnsrkqgmxqjaqaxgqyqm7fwyi96fk-binutils-2.31.1/bin/ld: /nix/store/s7fyj3h4z4h5r7mlwzhdvpk89kjmx785-kservice-5.58.0/lib/libKF5Service.so.5.58.0: undefined reference to `qt_version_tag@Qt_5.12'
  /nix/store/mgdjnsrkqgmxqjaqaxgqyqm7fwyi96fk-binutils-2.31.1/bin/ld: /nix/store/pll7y62wa1xzzvmhpjkd5jvjz2r2jqw2-kwidgetsaddons-5.58.0/lib/libKF5WidgetsAddons.so.5.58.0: undefined reference to `QCoreApplication::sendEvent(QObject*, QEvent*)@Qt_5'
  collect2: error: ld returned 1 exit status
  make[2]: *** [kstyle/config/CMakeFiles/breeze-settings5.dir/build.make:99: kstyle/config/breeze-settings5] Error 1
  make[1]: *** [CMakeFiles/Makefile2:645: kstyle/config/CMakeFiles/breeze-settings5.dir/all] Error 2
  make: *** [Makefile:130: all] Error 2
cannot build derivation '/nix/store/a7zm7zd5id30fk0spchqva8cxa0dgczn-kwin-5.15.5.drv': 1 dependencies couldn't be built
builder for '/nix/store/jx9z3sjln4aqb9n28ga6sq6n6qbl5625-python2.7-cheroot-6.5.6.drv' failed with exit code 1; last 10 log lines:
  0.98s teardown cheroot/test/test_ssl.py::test_tls_client_auth[0-True-localhost-pyopenssl]
  =========================== short test summary info ============================
  XFAIL cheroot/test/test_conn.py::test_Chunked_Encoding
    Server does not correctly read trailers/ending of the previous HTTP request, thus the second request fails as the server tries to parse b'Content-Type: application/json\r\n' as a Request-Line. This results in HTTP status code 400, instead of 413Ref: https://github.com/cherrypy/cheroot/issues/69
  XFAIL cheroot/test/test_conn.py::test_598
    Sometimes this test fails due to low timeout. Ref: https://github.com/cherrypy/cherrypy/issues/598
  XFAIL cheroot/test/test_core.py::test_parse_uri_unsafe_uri
    Fails on Python 2
  SKIPPED [1] cheroot/test/test_server.py:117: Abstract sockets don't work currently
  ==== 4 failed, 99 passed, 1 skipped, 3 xfailed, 1 warnings in 83.86 seconds ====
cannot build derivation '/nix/store/070040jmxzm7xby779bzs5xq9g8vcfd8-python2.7-cherrypy-17.4.2.drv': 1 dependencies couldn't be built
```
Further, is it really correct that this package has 4 different versions of `gst_plugins_good` and `evince` in its dependencies?

**Comment by peterhoeg** [comment]
We should probably move that taglib change to staging instead.

**Comment by peterhoeg** [comment]
And here it is without the taglib change which is in its own PR.

**Comment by peterhoeg** [comment]
@GrahamcOfBorg build strawberry

**Pull request merged**